docs/fsharp/tutorials/fsharp-interactive/index.md

```diff
@@ -13,102 +13,233 @@ To run F# Interactive from the console, run `dotnet fsi`. You will find `dotnet
+
+The following example shows how to get and use command-line arguments:
 
 ```fsharp
-#r "nuget: FSharp.Data, 3.3.2"
+let args = fsi.CommandLineArgs
+
+for arg in args do
+    printfn "%s" arg
 ```
 
-or in a more explicit fashion:
+When evaluated, it prints all arguments. The first argument is always the name of the script that is evaluated:
+
+```dotnet
+dotnet fsi Script1.fsx hello world from fsi
+Script1.fsx
+hello
+world
+from
+fsi
+```
+
+Note that you can also use `System.Environment.GetCommandLineArgs()` to access the same arguments.
+
+## F# Interactive directive reference
+
+The `#r` and `#load` directives seen previously are only available in F# Interactive. There are several directives only available in F# Interactive:
+
+|Directive|Description|
+|---------|-----------|
+|`#r "nuget:..."`|References a package from Nuget|
+|`#r "assembly-name.dll"`|References an assembly on disk|
+|`#load "file-name.fsx"`|Reads a source file, compiles it, and runs it.|
+|`#help`|Displays information about available directives.|
+|`#I`|Specifies an assembly search path in quotation marks.|
+|`#quit`|Terminates an F# Interactive session.|
+|`#time "on"` or `#time "off"`|By itself, `#time` toggles whether to display performance information. When it is `"on"`, F# Interactive measures real time, CPU time, and garbage collection information for each section of code that is interpreted and executed.|
+
+When you specify files or paths in F# Interactive, a string literal is expected. Therefore, files and paths must be in quotation marks, and the usual escape characters apply. You can use the `@` character to cause F# Interactive to interpret a string that contains a path as a verbatim string. This causes F# Interactive to ignore any escape characters.
+
+## Interactive and compiled preprocessor directives
+
+When you compile code in F# Interactive, whether you are running interactively or running a script, the symbol **INTERACTIVE** is defined. When you compile code in the compiler, the symbol **COMPILED** is defined. Thus, if code needs to be different in compiled and interactive modes, you can use these preprocessor directives for conditional compilation to determine which to use. For example:
 
 ```fsharp
-#r "nuget: FSharp.Data, Version=3.3.2"
+#if INTERACTIVE
+// Some code that executes only in FSI
+// ...
+#endif
 ```
 
+## Using F# Interactive in Visual Studio
+
+To run F# Interactive through Visual Studio, you can click the appropriate toolbar button labeled **F# Interactive**, or use the keys **Ctrl+Alt+F**. Doing this will open the interactive window, a tool window running an F# Interactive session. You can also select some code that you want to run in the interactive window and hit the key combination **Alt+Enter**. F# Interactive starts in a tool window labeled **F# Interactive**. When you use this key combination, make sure that the editor window has the focus.
```

Comment:
This could probably use a small image. What do you think?

Reply:
I'm leaning towards not. It was suggested a few years ago that I not include screenshots just because of how often the UI for VS ends up changing.